### ISSUE TITLE
Incorrect linter info msg

### ISSUE THREAD

**Issue by lowrider59** [nvuillam/npm-groovy-lint#264]
When we have a method like:

```
public someType someMethod() {
  synchronized (someObject) {
    doSomething();
    return something;
  }
}

```
we get an output like:

` nnn    info     The method someMethod in class SomeClass is missing an explicit return  ImplicitReturnStatement
`

I think that's actually incorrect :)

**Comment by nvuillam** [comment]
@lowrider59 this is a CodeNarc issue, not a npm-groovy-lint one ^^

You may disable the rule in this file or on this specific piece of code using inline-disable comments :)

https://nvuillam.github.io/npm-groovy-lint/#disabling-rules-in-source

**Comment by vallon** [comment]
I believe it is complaining that your "synchronized" call is being used as the return value of the function.  The return that is there is returning from the closure (the nested block), not the function.

Try:

```
public someType someMethod() {
  return synchronized (someObject) {
    doSomething();
    return something;
  }
}
```
